src/Tools/dotnet-monitor/Extensibility/FolderExtensionRepository.cs

```diff
@@ -28,15 +29,42 @@ public FolderExtensionRepository(IFileProvider fileSystem, ILoggerFactory logger
 
         public override bool TryFindExtension(string extensionName, out IExtension extension)
         {
-            IDirectoryContents extensionDir = _fileSystem.GetDirectoryContents(extensionName);
+            string extensionPath = string.Empty;
+
+            bool isDotnetToolsLocation = _targetFolder == HostBuilderSettings.ExtensionDirectoryPath;
```

Comment:
I think you should have a separate ExtensionRepository implementation for the dotnet tools folder since its algorithm is very much different than the FolderExtensionRepository implementation.